lib/ProductOpener/APITest.pm

```diff
@@ -125,6 +126,32 @@ sub create_user ($ua, $args_ref) {
 	return;
 }
 
+sub create_product ($ua, $product_feilds) {
+
+	my %fields = (
+	code => '200000000099',
+	lang => "en",
+	product_name => "Test-75ml",
+	generic_name => "Tester",
+	quantity => "75 ml",
+	link => "https://github.com/openfoodfacts/openfoodfacts-server",
+	expiration_date => "test",
+	ingredients_text => "apple, milk",
+	origin => "france",
+	serving_size => "10g",
+	packaging_text => "no",
+	".submit" => "submit"
+	);
+
+	while (my ($key, $value) = each %{$product_feilds}) {
+		$fields{$key} = $value;
+	}
+	
+	my $response = $ua->post("http://world.openfoodfacts.localhost/cgi/product_jqm2.pl", Content => \%fields,);
```

Comment:
I think this call is to edit a product.

To add a product you should  first add it with a "code" parameter using a post on "cgi/product.pl".

Use you dev instance to add a product (see left column on the search page) with developpers tools opened and see the request made in "network".

Reply:
product.pl is the web form, product_jqm2.pl is the API (we'll have to rename it to something better someday!)